test/testing_utils.dart

```diff
@@ -0,0 +1,224 @@
+  await waitForAngular(driver);
+  if (removeOverlay) {
+    await sleep10();
+    await dismissOverlay(driver);
+  }
+  print(
+      "loginUserWithCredentials: Logged in as user " + username + " pw: " + pw);
+}
+
+Future dismissOverlay(WebDriver driver) async {
+  var continueBtn;
+  try {
+    await waitForAngular(driver);
+    continueBtn = await driver.findElement(const By.cssSelector(
+        'product-module > cutout-shadow-overlay > div > div.cutout'));
+  } on NoSuchElementException catch (e) {
+    print("dismissOverlay: overlay element not found.");
+  } catch (exception, stackTrace) {
+    print("Exceptions caught");
+    print(exception);
+    print(stackTrace);
+  } finally {
+    if (elementExists(continueBtn)) {
+      await continueBtn.click();
+    }
+    print("dismissOverlay: dismissed cutout-shadow-overlay.");
+    await waitForAngular(driver);
+  }
+}
+
+Future<bool> isLoggedIn(WebDriver driver) async {
+  var profilePicture = null;
+  bool loggedIn = false;
+  try {
+    await waitForAngular(driver);
+    profilePicture = await driver.findElement(const By.cssSelector(
+        'nav > div > div > div.menu-item--profile > a > account-picture'));
+  } on NoSuchElementException catch (e) {
+    print("isLoggedIn: Logged-in-profile element not found.");
+  } catch (exception, stackTrace) {
+    print("Exceptions caught");
+    print(exception);
+    print(stackTrace);
+  } finally {
+    loggedIn = elementExists(profilePicture);
+    return loggedIn;
+  }
+}
+
+Future logoutUser(WebDriver driver) async {
+  bool loggedIn = await isLoggedIn(driver);
+  var profileBtn;
+  var logoutBtn;
+  if (loggedIn) {
+    try {
+      await waitForAngular(driver);
+      profileBtn = await driver.findElement(
+          const By.cssSelector('div.menu-item--profile > a > account-picture'));
+      logoutBtn = await driver.findElement(const By.xpath(
+          '*//div[contains(@class, "menu-item")]/a/span[contains(text(), "Kilépés")]'));
+    } on NoSuchElementException catch (e) {
+      print("logoutUser: Logout element(s) not found.");
+    } catch (exception, stackTrace) {
+      print("Exceptions caught:");
+      print(exception);
+      print(stackTrace);
+    } finally {
+      await waitForAngular(driver);
+      if (elementExists(profileBtn) && elementExists(logoutBtn)) {
+        await profileBtn.click();
+        await logoutBtn.click();
+      }
+      print("logoutUser: Logged out");
+    }
+  } else {
+    print("logoutUser: User wasnt logged in");
+  }
+}
+
+Future clickMegvasarlom(WebDriver driver, String productSlug) async {
+  await navigateToPage(driver, productSlug);
+  await waitForAngular(driver);
+  await sleep5();
+  var btnBuy = await driver.findElement(const By.xpath(
+      "*//div[contains(@class, 'package-details')]/material-button/material-ripple"));
+  await btnBuy.click();
+  await waitForAngular(driver);
+  await sleep5();
+}
+
+Future<bool> checkForConsoleErrors(WebDriver driver, String logLevel) async {
```

Comment:
In such cases it is a good practice to provide a sensible default for the `logLevel` parameter (like `SEVERE`)